packages/catalog-process/src/services/readModelService.ts

```diff
@@ -428,6 +428,45 @@ export function readModelServiceBuilder(
       return result.data;
     },
 
+    async listAgreementsForEServiceDescriptor({
```

Comment:
why not using the `listAgreements`?

Reply:
The `listAgreements` currently doesn't have a `descriptorId` (or `descriptorIds`) parameter. So I decided to create a custom function, given the situation of the `getEservices`, where the approach of adding more and more parameters has complicated the maintenance of the query. If you are confident that we would not end up in the same situation, I could add a new parameter to the `listAgreements` and reuse that function.

Reply:
no guarantees 😄 
but I would think about refactoring if and when it becomes an actual problem

Reply:
Updated